schemas/compat-data.schema.json

```diff
@@ -39,6 +39,12 @@
             "required": ["type", "name"]
           }
         },
+        "implementation_url": {
+          "type": "string",
+          "format": "uri",
+          "pattern": "^https:\/\/(trac.webkit.org\/changeset\/|hg.mozilla.org\/mozilla-central\/rev\/|chromium.googlesource.com\/chromium\/src\/|crrev.com\/)",
```

Comment:
This regex is a bit complex. Can we use [a `oneOf` array](https://json-schema.org/understanding-json-schema/reference/combining.html#oneof) to provide a set of patterns for each engine/vendor? Something like this, I think:

```
"oneOf": [
            {
              "pattern": "^https://trac\.webkit\.org/changeset/",
            },
            {
              "pattern": "^https://hg\.mozilla\.org\/mozilla-central",
            },
          ]
```